teps/0044-composing-tasks-with-tasks.md

```diff
@@ -0,0 +1,179 @@
+---
+status: proposed
+title: Composing Tasks with Tasks
+creation-date: '2021-01-22'
+last-updated: '2021-01-22'
+authors:
+- '@bobcatfish'
+---
+
+# TEP-0044: Composing Tasks with Tasks
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [PipelineResources](#pipelineresources)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+  - [Overlap with TEP-0046](#overlap-with-tep-0046)
+- [Requirements](#requirements)
+- [References (optional)](#references-optional)
+
+## Summary
+
+This TEP describes a gap in composability that isn't currently fully solved by Tasks or Pipelines
+(tho PipelineResources do solve some of it!). 
+
+Ideally by addressing the issues described here we'd make it possible to create more efficient
+Pipelines and make it easier to use a Task easily via a TaskRun, and we'd address some of
+[the problems PipelineResources are trying to solve](https://docs.google.com/document/d/1KpVyWi-etX00J3hIz_9HlbaNNEyuzP6S986Wjhl3ZnA/edit#)
+making it more clear whether or not they are an abstraction we want to keep.
+
+[#1673: Pipeline Resources Redesign](https://github.com/tektoncd/pipeline/issues/1673)
+
+## Motivation
+
+Currently the only way to combine Tasks together is in a Pipeline. If you combine
+Tasks in a Pipeline and they need to share data (beyond a simple
+[result](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-results)),
+you'll need to provision a PVC or do some other similar, cloud specific thing,
+to [make a volume available](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md#specifying-volumesources-in-workspaces)
+that can be shared between them.
+
+Some problems with this:
+* PVCs add additional overhead, both in speed and in management (somewhat
+  mitigated by
+  [letting tekton create and delete volumes for you](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md#volumeclaimtemplate)
+  but the PVCs still slow down the overall execution.
+  * Sometimes you just want to run a Task with a teeny bit more functionality,
+    e.g. you want to use a task to build images, but you want to clone source
+    with git first, and it seems like overkill to have to construct a pipeline
+    deal with persistent volume claims
```

Comment:
I am still fuzzy about this one. "I want to build an image but I want to clone the source code first", is _two units_ for me, thus, make sense to be a pipeline. 
The fact that it adds overhead at runtime is one thing. The fact that it makes it more verbose (maybe) is another.

We already have a way to compose `Task` today, and it is called `Pipeline`. How pipeline run, with what overhead is one problem to tackle, make the authoring less verbose/more intuitive might be another. Not sure adding another level of compos-ability something we wanna do. 

_I am thinking outlood._

Reply:
ill just remove this